src/pymovements/dataset/dataset_files.py

```diff
@@ -210,94 +212,46 @@ def load_gaze_files(
-        )
-
-        if not preprocessed:
-            gaze_data = gaze_data.rename(definition.column_map)
-
-        # Add fileinfo columns to dataframe.
-        gaze_data = add_fileinfo(
-            definition=definition,
-            df=gaze_data,
-            fileinfo=fileinfo_row,
-        )
-
         if preprocessed and extension == 'feather':
             # Preprocessed data already has tuple columns.
-            gaze_df = GazeDataFrame(
-                gaze_data,
-                experiment=definition.experiment,
-                trial_columns=definition.trial_columns,
+            gaze_df = load_gaze_file(
+                filepath=filepath,
+                fileinfo_row=fileinfo_row,
+                definition=definition,
+                preprocessed=preprocessed,
+                custom_read_kwargs=definition.custom_read_kwargs,
             )
 
-        elif preprocessed and extension in {'csv', 'txt', 'tsv'}:
-            time_column = None
-            distance_column = None
-
-            # Time column is always available in every initialized GazeDataFrame.
-            time_column = 'time'
-
-            if 'distance' in gaze_data.columns:
-                distance_column = 'distance'
-
-            pixel_columns: list[str] = list(
-                set(GazeDataFrame.valid_pixel_position_columns) & set(gaze_data.columns),
-            )
-            position_columns: list[str] = list(
-                set(GazeDataFrame.valid_position_columns) & set(gaze_data.columns),
-            )
-            velocity_columns: list[str] = list(
-                set(GazeDataFrame.valid_velocity_columns) & set(gaze_data.columns),
-            )
-            acceleration_columns: list[str] = list(
-                set(GazeDataFrame.valid_acceleration_columns) & set(gaze_data.columns),
-            )
-
-            gaze_df = GazeDataFrame(
-                gaze_data,
-                experiment=definition.experiment,
-                trial_columns=definition.trial_columns,
-                time_column=time_column,
-                distance_column=distance_column,
-                pixel_columns=pixel_columns,
-                position_columns=position_columns,
-                velocity_columns=velocity_columns,
-                acceleration_columns=acceleration_columns,
-            )
         else:
             # Create GazeDataFrame
-            gaze_df = GazeDataFrame(
-                gaze_data,
-                experiment=definition.experiment,
-                trial_columns=definition.trial_columns,
-                time_column=definition.time_column,
-                distance_column=definition.distance_column,
-                pixel_columns=definition.pixel_columns,
-                position_columns=definition.position_columns,
-                velocity_columns=definition.velocity_columns,
-                acceleration_columns=definition.acceleration_columns,
+            gaze_df = load_gaze_file(
```

Comment:
You can leave out the `if-else` here, as both calls are identical.

Reply:
done